### PR TITLE
Incorrect login password bug

### PR DESCRIPTION
**Pull request by pmyip3**
# Description 
Login fields show a green checkmark despite the wrong password. Toast popup showing "incorrect email/password". Needs a fix in the login field validation.
Also modified the register form to make it consistent. 

# Type of change 
- [x] Bug fix
- [ ] New feature
- [ ] Refactoring
- [ ] Documentation 

# Problem
See below
image
image

# Solution description
Remove the toast and have validation in the login field
Change the login form to use formik, which support above the feature 

# Tests 
All unit tests pass? N/A

Builds successfully? Yes

Attach screenshot of the expected result (e.g. UI of the website/ postman api call result/ database changes)
image
image

Provide steps in detail for reviewers to reproduce the expected result of the code changes.
Test with the register and login functionality 

# Relevant document/ link (if any) 
https://stackoverflow.com/questions/56809526/handling-errors-from-api-with-formik
https://formik.org/docs/api/formik

# Risk
Low

### CODE REVIEW

**Review by kdav108 — CHANGES_REQUESTED** [review]
Nice clean code! 

Just one UI suggestion, the "incorrect email/password" message appears underneath the password field:

Screen Shot 2021-09-16 at 9 35 43 pm

It's a bit odd to have it appear there if the issue is with the email (like in the image above). 

Unfortunately the api doesn't give us enough detail on whether the issue is with password or email, and hence I'm wondering if we can make this error appear either under the Login button or above the "Email address" title. What do you think?

**Comment by pmyip3** [comment]
> Nice clean code!
> 
> Just one UI suggestion, the "incorrect email/password" message appears underneath the password field:
> 
> Screen Shot 2021-09-16 at 9 35 43 pm
> 
> It's a bit odd to have it appear there if the issue is with the email (like in the image above).
> 
> Unfortunately the api doesn't give us enough detail on whether the issue is with password or email, and hence I'm wondering if we can make this error appear either under the Login button or above the "Email address" title. What do you think?

I think it makes sense to put a red box above the email address, but not sure if it's consistent with the current error handling? As we have the pop-up error message, error message on specific fields already. Or maybe we can simply also turn the email input fields red?

**Comment by cyr17** [comment]
> > Nice clean code!
> > Just one UI suggestion, the "incorrect email/password" message appears underneath the password field:
> > Screen Shot 2021-09-16 at 9 35 43 pm
> > It's a bit odd to have it appear there if the issue is with the email (like in the image above).
> > Unfortunately the api doesn't give us enough detail on whether the issue is with password or email, and hence I'm wondering if we can make this error appear either under the Login button or above the "Email address" title. What do you think?
> 
> I think it makes sense to put a red box above the email address, but not sure if it's consistent with the current error handling? As we have the pop-up error message, error message on specific fields already. Or maybe we can simply also turn the email input fields red?

yes have both of them as red

File: client/src/actions/auth.js [hunk 1]
```diff
@@ -31,7 +31,13 @@ export const login = (authData) => async (dispatch) => {
     dispatch({
       type: LOG_IN_FAIL,
     });
-    dispatch(errorActionGlobalCreator(error));
+    // only show pop up error if it's not a client error, otherwise, show the error on the form instead
+    if (error.response.status === 400) {
+      // assuming the only client error is 'Incorrect email/passord'
+      setFieldError('password', 'Incorrect email/ password');
```
Comment (ssha0054) [inline client/src/actions/auth.js:37]: Can remove the space before password in `email/ password`

**Review by jfan0010 — APPROVED** [review]
Tested. LGTM

File: client/src/components/auth/authComponents/LoginForm.jsx [hunk 2]
```diff
@@ -1,79 +1,94 @@
...
 const LoginForm = () => {
   const dispatch = useDispatch();
-  const [inputs, setInputs] = useState({
-    email: '',
-    password: '',
+
+  const teamInfoSchema = yup.object({
+    email: yup
+      .string()
+      .required('Please Enter your Email'),
```
Comment (vgoh1) [inline client/src/components/auth/authComponents/LoginForm.jsx:19]: Can we have only have the first word with the first uppercase? Eg. 'Please enter your email'

File: client/src/components/auth/authComponents/LoginForm.jsx [hunk 3]
```diff
@@ -1,79 +1,94 @@
...
+    password: yup
+      .string()
+      .required('Please Enter your password'),
```
Comment (vgoh1) [inline client/src/components/auth/authComponents/LoginForm.jsx:22]: 'Please enter your password'

Comment (vgoh1) [reply client/src/components/auth/authComponents/LoginForm.jsx:22]: And to add, if possible, name here should be lowercase in the error message:
image

**Review by rpra0006 — APPROVED** [review]
Functionality works for me

**Review by aksh0001 — APPROVED** [review]

**Pull request merged**